numba/np/polynomial.py

```diff
@@ -57,3 +58,116 @@ def roots_impl(p):
             return roots
 
     return roots_impl
+
+
+@register_jitable
+def _trimseq(seq):
```

Comment:
No need for using a `@register_jitable` here. You can put the code inside the `pu.trimsec` overload and just call it inside other overloads.